Special auto click

Razharas opened the pull request:
Adds automatic mode for guns
Autoclick? What? Never heard
Made specially for gatling lasergun
:cl:
rscadd: some guns now can shoot as long as your mouse button is pressed, moving the mouse changes the target accordingly
/:cl:

Razharas committed:
Adds automatic mode for guns

Adds automatic mode for guns
Autoclick? What? Never heard

Razharas committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into SpecialAutoClick

TechnoAlchemisto commented:
pew pew

KazeEspada commented:
Fucking russian would make a machine gun

Core0verload commented on code/_onclick/autoclick.dm:
```diff
@@ -0,0 +1,29 @@
+/obj/item/proc/CanItemAutoclick(object, location, params)
+
+/obj/item/weapon/gun/CanItemAutoclick(object, location, params)
+	. = 1
```
Wouldn't that make all the guns full auto?

Razharas replied:
It would, i only added the usability while the balance of this is up to discussion

Core0verload replied:
Well, it would be !!!!!!!!FUN!!!!!!!! to merge it that way, but no.

GunHog commented:
Delicious laser spam.

Razharas committed:
Easiest way to balance guns

Easiest way to balance guns with var

Razharas committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into SpecialAutoClick

Razharas committed:
Makes click catcher into a greed of images

Makes click catcher into a greed of images
Now you can swipe through darkness and it works perfectly

duncathan commented on code/_onclick/autoclick.dm:
```diff
@@ -0,0 +1,29 @@
+/mob/living/CanMobAutoclick(object, location, params)
+	var/obj/item/h = get_active_hand()
+	if(h)
+		. = h.CanItemAutoclick(object, location, params)
```
make it `return h.CanItemAutoClick(object, location, params)` instead

Razharas replied:
I heard direct returns are slower than default return value setting

duncathan replied:
yeah we tested it; turns out we were wrong

duncathan replied:
explicit returns are faster for this

RemieRichards replied:
well we're not wrong, it just turns out it's RIDICULOUSLY inconsistent, to where there's no point bothering with it or not.

duncathan commented on code/_onclick/autoclick.dm:
```diff
@@ -0,0 +1,32 @@
+	var/automatic = 0 //can gun use it, 0 is no, anything above 0 is the delay between clicks in ds
+
+/obj/item/weapon/gun/CanItemAutoclick(object, location, params)
+	. = automatic
```
Make it `return automati ` instead

Razharas replied:
Same as above

duncathan commented on code/_onclick/click.dm:
```diff
@@ -323,19 +323,29 @@
 	screen_loc = "CENTER-7,CENTER-7"
 
+/obj/screen/click_catcher/proc/MakeGreed()
+	var/list/L = list()
```
Instead of making a new car, use .:
`. = list()`

Razharas replied:
Ok

duncathan commented on code/_onclick/click.dm:
```diff
@@ -323,19 +323,29 @@
+			var/obj/screen/click_catcher/CC = new()
+			CC.screen_loc = "NORTH-[i],EAST-[j]"
+			L.Add(CC)
+	return L
```
then instead of `L.Add(CC)` do `. += CC`
then remove the explicit return

Razharas replied:
Ok

Razharas committed:
Some tiny optimizations

Some tiny optimizations

Razharas committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into SpecialAutoClick

KorPhaeron commented:
I stole this code
